nautobot/core/ui/object_detail.py

```diff
@@ -811,7 +809,7 @@ def get_data(self, context):
 
         data = {}
 
-        if isinstance(instance, TreeModel):
+        if isinstance(instance, TreeModel) and self.label is None:
```

Comment:
I get what this is doing but it's not super-obvious. Would it be better to do something like `and (self.fields == "__all__" or "_hierarchy" in self.fields)`?